test_aiodataloader.py

```diff
@@ -1,36 +1,45 @@
+from collections.abc import Callable, Coroutine
 import pytest
 from asyncio import gather
 from functools import partial
 from pytest import raises
-
+from typing import Dict, List, Optional, Tuple, TypeVar
 from aiodataloader import DataLoader
 
 pytestmark = pytest.mark.asyncio
 
+
+T1 = TypeVar("T1")
+T2 = TypeVar("T2")
+
+
 async def do_test():
     return True
 
 
-def id_loader(**options):
+def id_loader(
+    *, resolve: Optional[Callable[..., Coroutine]] = None, **dl_kwargs
+) -> Tuple[DataLoader, List]:
     load_calls = []
 
-    async def default_resolve(x):
+    async def default_resolve(x: T1) -> T1:
         return x
 
-    resolve = options.pop('resolve', default_resolve)
+    if resolve is None:
+        resolve = default_resolve
 
-    async def fn(keys):
+    async def fn(keys: List) -> List:
         load_calls.append(keys)
         return await resolve(keys)
```

Comment:
I get a `"None" not callable` type error here.  Maybe get rid of the redefinition of resolve above, and make this:
```
return await (resolve or default_resolve)(keys)
```

Reply:
Hmm this should be narrowed by `if resolve is None` and I'm not getting that error, which type inference engine are you using (and what version)?

Reply:
(that said your suggestion is also fine so if it makes it work in more linters then that's probably the better option anyway)